knative-operator/pkg/monitoring/sources/source_deployment_discovery_controller.go

```diff
@@ -2,23 +2,31 @@ package sources
 
 import (
 	"context"
+	"os"
+	"strconv"
 
 	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/common"
 	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/monitoring"
+	okomon "github.com/openshift-knative/serverless-operator/openshift-knative-operator/pkg/monitoring"
 	v1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/operator/pkg/apis/operator/v1alpha1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
-	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
-	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-var log = common.Log.WithName("source-deployment-discovery-controller")
+var (
+	generateSourceServiceMonitorsEnvVar = "GENERATE_SERVICE_MONITORS_FOR_SOURCES_BY_DEFAULT"
```

Comment:
Why the extension `BY_DEFAULT` ? How can this be overridden ? (e.g. a user sets this variable, but for certain sources it does not want a ServiceMonitor to be created ?

If this variable triggers to _always_ create a ServiceMonitor for a Source, then I would ommit the `BY_DEFAULT` suffix.

Reply:
This can be overridden via a subscription update check comments above. I can ommit BY_DEFAULT.